src/handlers/pr_tracking.rs

```diff
@@ -137,15 +174,20 @@ AND CARDINALITY(r.assigned_prs) < LEAST(COALESCE(r.max_assigned_prs,1000000));";
 /// Ensures no accidental PR duplicates.
 async fn upsert_pr_into_workqueue(
     db: &DbClient,
-    user_id: u64,
+    user: &User,
     pr: u64,
 ) -> anyhow::Result<u64, anyhow::Error> {
+    // Ensure the user has entry in the `users` table
+    record_username(db, user.id, &user.login)
+        .await
+        .context("failed to record username")?;
```

Comment:
Here there is now a "hidden" a second SQL query where I expect to only have one. `upsert_pr_into_workqueue` says to just "Add a PR to the workqueue of a team member", we are sort of breaking this promise.

Maybe call `record_username()` just before (at the cost of calling everywhere)?

Reply:
I think that it's better if the function works in a self-contained way, rather than not working if you don't call a specific other function before it. We don't call it in a loop outside tests, it's always called incrementally, and the code that calls it always called `record_username` before it anyways, so there wouldn't be any change.

But mainly, I want to nuke this whole code in a follow-up PR, so.. :laughing: